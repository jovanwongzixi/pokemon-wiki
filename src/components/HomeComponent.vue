<template>
    <div>
        <iframe src="https://giphy.com/embed/6nWT5GyxsmAoM" width="200" height="168" frameBorder="0" class="giphy-embed" allowFullScreen></iframe>
        <p><a href="https://giphy.com/gifs/6nWT5GyxsmAoM">via GIPHY</a></p>
    </div>
        <p>This is a mini pokemon wiki site for me to practise my VueJS skills, along with unit testing using Vitest. Design to display pokemon info heavily inspired by Bulbapedia. Click on the Pokemon tab to start searching for a Pokemon!!</p>
</template>

<style scoped>
div {
    display: flex;
    flex-direction: column;
    align-items: center;
}
p {
    margin-top: 0.2rem;
    font-weight: 500;
}
</style>