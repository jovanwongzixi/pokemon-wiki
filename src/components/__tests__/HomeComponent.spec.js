import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import HomeComponent from '../HomeComponent.vue'

describe('HomeComponent', () => {
    it('renders properly', () => {
        const wrapper = mount(HomeComponent)
        expect(wrapper.text()).toMatchSnapshot()//toContain('This is a mini pokemon wiki site for me to practise my VueJS skills, along with unit testing using Vitest. Design to display pokemon info heavily inspired by Bulbapedia. Click on the Pokemon tab to start searching for a Pokemon!!')
    })
})