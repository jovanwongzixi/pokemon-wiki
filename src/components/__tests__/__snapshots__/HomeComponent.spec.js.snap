// Vitest Snapshot v1

exports[`HomeComponent > renders properly 1`] = `"via GIPHYThis is a mini pokemon wiki site for me to practise my VueJS skills, along with unit testing using Vitest. Design to display pokemon info heavily inspired by Bulbapedia. Click on the Pokemon tab to start searching for a Pokemon!!"`;
